### PR TITLE
Decouple source data from the detection/classification messages.

### PR DESCRIPTION
**Pull request by peci1**
Following the discussion in #43, mainly:

> Decouple from data: Some messages here include a field carrying some source data. I understand it is very practical on one hand (we do it too in our custom messages), but on the other hand it generates issues like #20 (and possibly future issues like sensor_msgs/PointCloud support etc.). And these have no good solution with the data inside the message except for composition on a higher level. I propose to remove the source data fields from vision_msgs, and possibly leave a note in the message definition that the user should either use a message synchronizer to get the pairs of image-detection, or create a custom type that contains both the image (cloud...) and the detection message. It's easy to use topic_tools/relay_field to extract the desired parts afterwards. There are situations where you do not want to relate the detection to the whole image, but you want to e.g. have the cutout from the image where the detection occured. This would work the same in case 1 detection per image is published (you'd just publish the cutout on a new topic). But it would become a bit more complicated in case the detector can return multiple detections per image and you would use DetectionArray. Here, matching by header would no longer be enough. Or - if taken rigorously - it could be, because making a cutout from an image formally invalidates its frame_id. So each cutout should have its own frame_id, and then you could make the detection-image mapping. Some more rationale: someone might want to run detections on an organized 3D pointcloud from a depth camera and return 2D detections (because he's e.g. only interested in bounding boxes in the RGB in the end). And a practical issue we faced: on a low-bandwidth link, we wanted to send detections, but as I mentioned earlier, the image is already a part of our messages (similarly to how it is done here now). But we figured out we do not want to transmit the image over the slow link. So we ended up not filling any values in the image fields, and now we have detections which lie to be detected on 0x0 px images. That doesn't really seem like a conceptual approach. Still, I'm not really 100% sure on this point, but it seems to me decoupling is the "more correct" way, but has its pros and cons. Bad thing is that using composition to create a DetectionsWithImagesArray message would not subsequently allow to use topic_tools/relay_field, because you would need to relay primitive-array fields, which I guess is not supported.

This PR drops dependency on sensor_msgs.

### CODE REVIEW

**Comment by peci1** [comment]
I don't have a ROS2 compilation pipeline at hand, so please check if it compiles before merging.

File: README.md [hunk 1]
```diff
@@ -42,22 +42,19 @@ classifier information.
 
 ## Messages
 
-  * Classification2D and Classification3D: pure classification without pose
+  * Classification: pure classification without pose
   * Detection2D and Detection3D: classification + pose
-  * XArray messages, where X is one of the four message types listed above. A
+  * XArray messages, where X is one of the two message types listed above. A
```
Comment (mintar) [inline README.md:54]: Should this be "three" message types (Classification, Detection2D, Detection3D)?

File: msg/Classification.msg [hunk 2]
```diff
@@ -1,7 +1,7 @@
-# Defines a 2D classification result.
+# Defines a classification result.
 #
 # This result does not contain any position information. It is designed for
-#   classifiers, which simply provide class probabilities given a source image.
+#   classifiers, which simply provide class probabilities given an instance of source data.
```
Comment (mintar) [inline msg/Classification.msg]: Maybe add "(e.g., an image or a point cloud)"?

File: msg/Classification.msg [hunk 3]
```diff
@@ -10,6 +10,4 @@ std_msgs/Header header
 #   user-defined threshold.
 ObjectHypothesis[] results
 
-# The 2D data that generated these results (i.e. region proposal cropped out of
-#   the image). Not required for all use cases, so it may be empty.
-sensor_msgs/Image source_img
\ No newline at end of file
+# Source data that generated this classification are not a part of the message. If you need to access them, use an exact or approximate time synchronizer in your code, as this message's header should match the header of the source data.
```
Comment (mintar) [inline msg/Classification.msg]: Should we move the new comment to the README? Same for Detection2D and Detection3D.

File: msg/Detection3D.msg [hunk 4]
```diff
@@ -1,8 +1,7 @@
 # Defines a 3D detection result.
 #
 # This extends a basic 3D classification by including position information,
-#   allowing a classification result for a specific position in an image to
-#   to be located in the larger image.
+#   allowing a classification result for a specific position in data.
```
Comment (mintar) [inline msg/Detection3D.msg]: You removed the verb ("located"). Maybe rephrase the whole comment? Something like this:

"This extends a basic 3D classification by including the pose of the detected object."

**Review by mintar — COMMENTED** [review]
Looks good to me, except for some remarks on the comments!

File: README.md (hunk 1 above)
Comment (peci1) [reply README.md:54]: There's no ClassificationArray (and I think that's right, what would that be for?)

File: msg/Classification.msg (hunk 3 above)
Comment (peci1) [reply msg/Classification.msg]: I'll add it to README, but leave it here, because this place is IMO much more visible than the README.

File: msg/Detection3D.msg (hunk 4 above)
Comment (peci1) [reply msg/Detection3D.msg]: Yes, you're right, I was also unhappy about the sentence but haven't quickly come up with anything better.

**Review by SteveMacenski — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (mintar) [reply README.md:54]: Oh right. I just saw that you removed one message, so I assumed it should go from "four" to "three", but you're absolutely right - it should already have said "two" before.

**Review by mintar — APPROVED** [review]
